pkg/util/test/util.go

```diff
@@ -1887,6 +1893,47 @@ func validateCSITopologyFeatureGate(pod v1.Pod, topologyEnabled bool) error {
 	return nil
 }
 
+func validateCSISnapshotController(cluster *corev1.StorageCluster, pxImageList map[string]string, timeout, interval time.Duration) error {
```

Comment:
Hey I see you made a whole different function to validate CSI Snapshot Controller, I was wondering if we could just add this logic here https://github.com/libopenstorage/operator/blob/922d419739ffd3c3eda9ea07d413c37b088a3a6f/pkg/util/test/util.go#L1922 and just check if its enabled then do all this validation inside `validateCsiExtImages()`, because we already getting the deployment `px-csi-ext` info here, so could be an easy if statement to validate the images there as well like we do for the rest of `px-csi-ext` container images, what do you think?

Reply:
probably also make sense to improve this func for when all this CSI components enabled and disabled like we do for lets say Telemetry here https://github.com/libopenstorage/operator/blob/922d419739ffd3c3eda9ea07d413c37b088a3a6f/pkg/util/test/util.go#L2363, that is on my plate to do similar thing for all components as we are lacking validation for when they are disabled.

Reply:
the rest of the images are always there, and they are validated after csi deployment is ready. while when toggling this snapshot controller, we actually need retry and sleep to get updated deployment until old pods are deleted and new pods are created. don't want to add the retry logic there to make that function complex, having a separate validation for this snapshot controller is more straightforward imo.
I thought about doing it that way but ended up doing this..